File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/model/ContainerType.java
package dam.isi.frsf.utn.edu.ar.delivery.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class ContainerType implements Serializable
{

    @SerializedName("label")
    @Expose
    private String label;
    @SerializedName("imgURL")
    @Expose
    private String imgURL;
    @SerializedName("maxFlavors")
    @Expose
    private Long maxFlavors;
    @SerializedName("variableQuantityOfFlavors")
    @Expose
    private Boolean variableQuantityOfFlavors;
    private final static long serialVersionUID = -2667461013612687786L;
    @SerializedName("priceInCents")
    @Expose
    private int priceInCents;

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public ContainerType withLabel(String label) {
        this.label = label;
        return this;
    }

    public String getImgURL() {
        return imgURL;
    }

    public void setImgURL(String imgURL) {
        this.imgURL = imgURL;
    }

    public ContainerType withImgURL(String imgURL) {
        this.imgURL = imgURL;
        return this;
    }

    public int getPriceInCents() {
        return priceInCents;
    }

    public void setPriceInCents(int priceInCents) {
        this.priceInCents = priceInCents;
    }

    public ContainerType withPriceInCents(int priceInCents) {
        this.priceInCents = priceInCents;
        return this;
    }

    public Long getMaxFlavors() {
        return maxFlavors;
    }

    public void setMaxFlavors(Long maxFlavors) {
        this.maxFlavors = maxFlavors;
    }

    public ContainerType withMaxFlavors(Long maxFlavors) {
        this.maxFlavors = maxFlavors;
        return this;
    }

    public Boolean getVariableQuantityOfFlavors() {
        return variableQuantityOfFlavors;
    }

    public void setVariableQuantityOfFlavors(Boolean variableQuantityOfFlavors) {
        this.variableQuantityOfFlavors = variableQuantityOfFlavors;
    }

    public ContainerType withVariableQuantityOfFlavors(Boolean variableQuantityOfFlavors) {
        this.variableQuantityOfFlavors = variableQuantityOfFlavors;
        return this;
    }

}

File: mobile/app/src/main/java/dam/isi/frsf/utn/edu/ar/delivery/model/ContainerType.kt
package dam.isi.frsf.utn.edu.ar.delivery.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import dam.isi.frsf.utn.edu.ar.delivery.constants.appConstants

data class ContainerType(
        @SerializedName("label")
        @Expose
        var label: String = "",
        @SerializedName("imgURL")
        @Expose
        var imgURL: String = "",
        @SerializedName("maxFlavors")
        @Expose
        var maxFlavors: Long = 0L,
        @SerializedName("variableQuantityOfFlavors")
        @Expose
        var variableQuantityOfFlavors: Boolean = false,
        @SerializedName("priceInCents")
        @Expose
        var priceInCents: Int = 0
) : Parcelable {

    fun withLabel(label: String): ContainerType {
        this.label = label
        return this
    }

    fun withImgURL(imgURL: String): ContainerType {
        this.imgURL = imgURL
        return this
    }

    fun withPriceInCents(priceInCents: Int): ContainerType {
        this.priceInCents = priceInCents
        return this
    }

    fun withMaxFlavors(maxFlavors: Long): ContainerType {
        this.maxFlavors = maxFlavors
        return this
    }

    fun withVariableQuantityOfFlavors(variableQuantityOfFlavors: Boolean): ContainerType {
        this.variableQuantityOfFlavors = variableQuantityOfFlavors
        return this
    }

    val completeImgURL: String = appConstants.staticPath + "containerImages/" + imgURL

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<ContainerType> = object : Parcelable.Creator<ContainerType> {
            override fun createFromParcel(source: Parcel): ContainerType = ContainerType(source)
            override fun newArray(size: Int): Array<ContainerType?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
    source.readString(),
    source.readString(),
    source.readLong(),
    1 == source.readInt(),
    source.readInt()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(label)
        dest.writeString(imgURL)
        dest.writeLong(maxFlavors)
        dest.writeInt((if (variableQuantityOfFlavors) 1 else 0))
        dest.writeInt(priceInCents)
    }
}